core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -985,15 +985,23 @@ public void addFailedEngineListener(Engine.FailedEngineListener failedEngineList
     }
 
     public void updateBufferSize(ByteSizeValue shardIndexingBufferSize, ByteSizeValue shardTranslogBufferSize) {
+        Engine engine = engineUnsafe();
```

Comment:
I think this is wrong? we should change the config even if the engine is not yet started (a null engine doesn't always mean shard closed). I think we run into issues here in the past..